### PR TITLE
[GH-241] Unity: adds lun migration support

### PR DESCRIPTION
**Pull request by yong-huang**
Use move session to implement lun migration.

### CODE REVIEW

**Comment by Murray-LIANG** [comment]
Could you fix the pep8 failure in another PR?

File: storops/unity/resource/lun.py [hunk 1]
```diff
@@ -353,6 +358,43 @@ def thin_clone(self, name, io_limit_policy=None, description=None):
...
+            move_session = clz.get(self._cli, _id=move_session_id)
+            if move_session.state == enums.MoveSessionStateEnum.COMPLETED:
+                return True
+            return False
+
+        clz = storops.unity.resource.move_session.UnityMoveSession
+        is_compressed = self.is_data_reduction_enabled
+
+        try:
+            move_session = clz.create(self._cli, self, dest,
```
Comment (Murray-LIANG) [inline storops/unity/resource/lun.py]: What if the lun is in cg?

File: storops/unity/resource/lun.py [hunk 2]
```diff
@@ -353,6 +358,43 @@ def thin_clone(self, name, io_limit_policy=None, description=None):
...
+        if not self._is_move_session_supported(dest):
+            return False
+
+        interval = kwargs.pop('interval', 5)
+        timeout = kwargs.pop('timeout', 1800)
+
+        @retryz.retry(timeout=timeout, wait=interval, on_return=False)
+        def _do_check_move_session(move_session_id):
+            move_session = clz.get(self._cli, _id=move_session_id)
+            if move_session.state == enums.MoveSessionStateEnum.COMPLETED:
```
[the remark this answers is not in the document]
Comment (yong-huang) [reply storops/unity/resource/lun.py]: Fixed, will check FAILED and CANCELLED states.

File: storops/unity/resource/lun.py (hunk 1 above)
Comment (yong-huang) [reply storops/unity/resource/lun.py]: If lun in cg, REST will return 409 source or dest does not exist error, storops will raise UnityMigrationSourceDestNotExistsError exception.

**Pull request merged**